### PR TITLE
add stephist

### PR DESCRIPTION
**Pull request by Moelf**
# Description

helps #368

Looks stupid right now because I just copied the `Hist`, probably can be simplified, welcome comments

## Type of change

Delete options that do not apply:

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] Added an entry in NEWS.md (for new features and breaking changes)
- [x] Added or changed relevant sections in the documentation
- [ ] Added unit tests for new algorithms, conversion methods, etc.
- [ ] Added reference image tests for new plotting functions, recipes, visual options, etc.

### CODE REVIEW

**Comment by Moelf** [comment]
```julia
julia> f = stephist(rand(10000))
FigureAxisPlot()
```

image

File: src/stats/hist.jl [hunk 1]
```diff
@@ -8,6 +8,60 @@ function convert_arguments(P::Type{<:AbstractPlot}, h::StatsBase.Histogram{<:Any
...
+        offset = 0.0,
+        scale_to = nothing,
+
+        flip_labels_at = Inf,
+        label_color = theme(scene, :textcolor),
+        over_background_color = automatic,
+        over_bar_color = automatic,
+        label_offset = 5,
+        label_font = theme(scene, :font),
+        label_size = 20,
```
Comment (lazarusA) [inline src/stats/hist.jl:27]: maybe consider adding `linestyle`.

**Comment by Moelf** [comment]
if people are feeling okay with the implementation I can start adding News, Docs and test samples

**Comment by Moelf** [comment]
What's the core maintainers' opinion on picking up a histogram package as a dependency? I've grown tired of re-writing crippled binning logic in multiple places :(

**Comment by Moelf** [comment]
this is ready I think. Looks like we don't do reference image for this kind of non basic plots? I checked waterfall and tricontourf PRs.

Let me know if we want anything else

**Comment by jkrumbiegel** [comment]
> Looks like we don't do reference image for this kind of non basic plots? 

Would still be great if you added a reference image, could be one image with a couple of subplots that expose different settings (ideally thick lines so that the visual differ would pick up on it if anything broke).

Also, the docs build is currently broken because of an `automatic` value being passed too far into the GLMakie backend it seems?

**Comment by Moelf** [comment]
Do you have an example of PR adding references imagine?

**Comment by jkrumbiegel** [comment]
The PRs don't add the images themselves, just the code. You can check them in the ReferenceTests folder, all the @reference macros

**Comment by Moelf** [comment]
Am I doing it correctly with reference?

**Comment by Moelf** [comment]
can't seem to find which automatic is too far

**Comment by lazarusA** [comment]
@Moelf this looks like almost there. Maybe a final sprint to make it possible.

**Comment by Moelf** [comment]
yeah I just don't understand what `Automatic()` is getting propagated, since I pretty much took a subset from regular histogram

**Comment by Moelf** [comment]
https://github.com/MakieOrg/Makie.jl/actions/runs/4018541242/jobs/6904286091#step:7:613

I think we have made progress, now only the Reference saving is not working, due to what seems to be because it's trying to save Observable?

**Comment by SimonDanisch** [comment]
I cleaned up the attributes! I think you copied a few attributes from hist, without using them?
If I removed anything that wasn't supposed to be removed I can revert the commit ;) 
But the error was due to passing `label_font` to `stairs`, which doesn't support that attribute

**Comment by asinghvi17** [comment]
@cossio that'll be done in #2636

**Pull request merged**